include/mra/tasks/multiply.h

```diff
@@ -39,31 +39,49 @@ namespace mra{
 #ifndef MRA_ENABLE_HOST
       auto sends = ttg::device::forward();
       auto send_out = [&]<typename S>(S&& out){
-        sends.push_back(ttg::device::send<0>(key, std::forward<S>(out)));
+        sends.push_back(ttg::device::send<0>(out, std::forward<S>(out)));
       };
 #else
       auto send_out = [&]<typename S>(S&& out){
         ttg::send<0>(key, std::forward<S>(out));
       };
 #endif
 
+      // If nodeA and nodeB leaf:
+      // call the kernel
+      // If nodeA == non-leaf and nodeB = leaf:
+      //  forward B to leaf nodes in A
+      // If nodeB == non-leaf and nodeA = leaf:
+      //  forward A to leaf nodes in B
+
       if (t1.empty() || t2.empty()) {
         /* send out an empty result */
-        auto out = mra::FunctionsReconstructedNode<T, NDIM>(key, N);
-        mra::apply_leaf_info(out, t1, t2);
-        send_out(std::move(out));
+        if(!t1.empty()){
+          auto out = mra::FunctionsReconstructedNode<T, NDIM>(t1.key(), N, K, ttg::scope::Allocate);
+          mra::apply_leaf_info(out, t2);
+          send_out(std::move(out));
+        } else if (!t2.empty()) {
+          auto out = mra::FunctionsReconstructedNode<T, NDIM>(t2.key(), N, K, ttg::scope::Allocate);
```

Comment:
Why send an empty node? This is where we should send T2 to the children (because they already have the input from t1)

Reply:
I'm assuming you mean children of t1 ?

Reply:
It's the children of the current task, so the key.